internal/ingress/controller/template/template.go

```diff
@@ -588,6 +595,33 @@ func buildAuthResponseHeaders(proxySetHeader string, headers []string) []string
 	return res
 }
 
+func buildAuthUpstreamHeaders(proxySetHeader string, headers []string) []string {
+	res := []string{}
+
+	if len(headers) == 0 {
+		return res
+	}
+
+	for i, h := range headers {
+		res = append(res, fmt.Sprintf("set $authHeader%d '';", i))
+		res = append(res, fmt.Sprintf("%s '%s' $authHeader%d;", proxySetHeader, h, i))
+	}
+	return res
+}
+
+func buildAuthUpstreamLuaHeaders(proxySetHeader string, headers []string) []string {
```

Comment:
I don't get the reason of this function and why it receives proxySetHeader string, as this is not used